internal/juju/client.go

```diff
@@ -17,20 +18,13 @@ type Client struct {
 	Models modelsClient
 }
 
-func NewClient(config Configuration) (*Client, error) {
-	connr, err := connector.NewSimple(connector.SimpleConfig{
-		ControllerAddresses: config.ControllerAddresses,
-		Username:            config.Username,
-		Password:            config.Password,
-		CACert:              config.CACert,
-	})
-	if err != nil {
-		return nil, err
-	}
+type ConnectionFactory struct {
+	config Configuration
+}
 
-	conn, err := connr.Connect()
-	if err != nil {
-		return nil, err
+func NewClient(config Configuration) (*Client, error) {
+	cf := ConnectionFactory{
```

Comment:
Am I being too lazy with this? Should it be named `connectionFactory`?

Reply:
No, `cf` is fine in this context.